### PR TITLE
Runtime Retries Limit

### PR DESCRIPTION
**Pull request by ben-toogood**
Implement max 3 retries for local and k8s runtimes.

### CODE REVIEW

**Comment by asim** [comment]
Any reason its implemented as a field in the service rather than passed as an option and maybe returned back in metadata like status and error?

**Comment by asim** [comment]
Actually you either want retries as a CreateOption or in metadata. The Service struct should remain clean with only the service details itself.

**Comment by ben-toogood** [comment]
Thanks @asim. I'll amend this.

**Comment by ben-toogood** [comment]
@asim I've actioned this, just need to implement the micro side. Can you pls review.

File: runtime/options.go [hunk 1]
```diff
@@ -37,6 +39,13 @@ func WithType(t string) Option {
 	}
 }
 
+// WithMaxRetries sets the arg MaxRetries
+func WithMaxRetries(max int) Option {
```
Comment (asim) [inline runtime/options.go]: Just call with WithRetries and the option as Retries

**Comment by ben-toogood** [comment]
@asim updated based on feedback, I agree this is better

File: runtime/options.go [hunk 2]
```diff
@@ -87,6 +80,13 @@ func WithCommand(args ...string) CreateOption {
 	}
 }
 
+// WithRetires sets the max retries attemps
+func WithRetires(retries int) CreateOption {
```
Comment (asim) [inline runtime/options.go]: Retires => Retries

File: runtime/service.go [hunk 3]
```diff
@@ -58,9 +58,10 @@ func newService(s *Service, c CreateOptions) *service {
 			Env:  c.Env,
 			Args: args,
 		},
-		closed:  make(chan bool),
-		output:  c.Output,
-		updated: time.Now(),
+		closed:     make(chan bool),
+		output:     c.Output,
+		updated:    time.Now(),
+		maxRetries: c.Retries,
```
Comment (asim) [inline runtime/service.go:64]: retries

**Pull request merged**